cxx-checks/src/main/java/org/sonar/cxx/checks/FunctionCognitiveComplexityCheck.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Sonar C++ Plugin (Community)
+ * Copyright (C) 2010-2017 SonarOpenCommunity
+ * http://github.com/SonarOpenCommunity/sonar-cxx
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.cxx.checks;
+
+import org.sonar.check.Priority;
+import org.sonar.check.Rule;
+import org.sonar.check.RuleProperty;
+import org.sonar.cxx.api.CxxMetric;
+import org.sonar.cxx.parser.CxxGrammarImpl;
+import org.sonar.squidbridge.api.SourceFunction;
+import org.sonar.squidbridge.checks.ChecksHelper;
+import org.sonar.squidbridge.checks.SquidCheck;
+import com.sonar.sslr.api.AstNode;
+import com.sonar.sslr.api.Grammar;
+import org.sonar.squidbridge.annotations.ActivatedByDefault;
+import org.sonar.squidbridge.annotations.SqaleLinearWithOffsetRemediation;
+import org.sonar.cxx.tag.Tag;
+
+@Rule(
+  key = "FunctionCognitiveComplexity",
+  name = "Functions should not be too complex",
+  tags = {Tag.BRAIN_OVERLOAD},
+  priority = Priority.MAJOR)
+@ActivatedByDefault
+@SqaleLinearWithOffsetRemediation(
+  coeff = "1min",
+  offset = "10min",
+  effortToFixDescription = "per complexity point above the threshold")
+public class FunctionCognitiveComplexityCheck extends SquidCheck<Grammar> {
+
+  private static final int DEFAULT_MAX = 10;
```

Comment:
Default in other languages (Java and JS) is 15: https://github.com/SonarSource/SonarJS/blob/master/javascript-checks/src/main/java/org/sonar/javascript/checks/CognitiveComplexityFunctionCheck.java#L46

Reply:
Fixed.